src/coreComponents/mesh/generators/VTKFaceBlockUtilities.cpp

```diff
@@ -0,0 +1,633 @@
+  {
+    return pairToPairSig( p0 ) < pairToPairSig( p1 );
+  };
+  std::set< std::pair< int, int >, decltype( cmp ) > uniqueEdges( cmp );
+  for( std::vector< std::pair< int, int > > const & edges: allEdges )
+  {
+    uniqueEdges.insert( edges.cbegin(), edges.cend() );
+  }
+  GEOSX_ERROR_IF_NE_MSG( LvArray::integerConversion< localIndex >( uniqueEdges.size() ), num2dFaces, "Internal error: wrong number of edges in the fracture/fault mesh computation." );
+
+  array1d< localIndex > face2dToEdges;
+  face2dToEdges.reserve( num2dFaces );
+  for( std::pair< int, int > const & p: uniqueEdges )
+  {
+    face2dToEdges.emplace_back( pairToEdge( p, nodeToEdges ) );
+  }
+
+  return { face2dToEdges, elem2dToEdges };
+}
+
+/**
+ * @brief Compute the 2d faces (geometrical edges in 2d) to 2d elems (geometrical faces in 3d) mapping.
+ * @param[in] num2dFaces The number of 2d faces (geometrical edges in 3d).
+ * @param[in] face2dToEdges The 2d faces (geometrical edges in 2d) to edges mapping.
+ * @param[in] elem2dToEdges The 2d elem (geometrical faces in 3d) to edges mapping.
+ * @return The mapping.
+ */
+ArrayOfArrays< localIndex > compute2dFaceToElems2d( localIndex const num2dFaces,
+                                                    arrayView1d< localIndex const > face2dToEdges,
+                                                    ArrayOfArraysView< localIndex const > elem2dToEdges )
+{
+  ArrayOfArrays< localIndex > face2dToElems2d( num2dFaces );
+
+  // First, a mapping inversion
+  std::map< int, std::vector< int > > edgesToElems2d;
+  for( localIndex i = 0; i < elem2dToEdges.size(); ++i )
+  {
+    for( int const & e: elem2dToEdges[i] )
+    {
+      edgesToElems2d[e].push_back( i );
+    }
+  }
+  // Then we fill the 2d face -> 2d elems mappings using the 2d face -> edges and edges -> 2d elems chain.
+  for( int fi = 0; fi < num2dFaces; ++fi )
+  {
+    localIndex const & edge = face2dToEdges[fi];
+    std::vector< int > const & elems2d = edgesToElems2d[edge];
+    face2dToElems2d.resizeArray( fi, elems2d.size() );
+    std::copy( elems2d.begin(), elems2d.end(), face2dToElems2d[fi].begin() );
+  }
+
+  return face2dToElems2d;
+}
+
+void importFracture( string const & faceBlockName,
+                     vtkSmartPointer< vtkDataSet > vtkMesh,
+                     CellBlockManager & cellBlockManager )
+{
+  vtkFieldData * fieldData = vtkMesh->GetFieldData();
+  GEOSX_ERROR_IF( fieldData == nullptr, "Fault and fracture information is supposed to be defined in the VTK field data. None was found." );
+
+  FaceData const faceData( fieldData, faceBlockName );
+
+  vtkIdType const num2dElements = faceData.num2dElements();
+  localIndex const num2dFaces = computeNum2dFaces( vtkMesh, faceData );
+
+  internal::ElementToFace const elemToFaces( cellBlockManager.getCellBlocks() );
+
+  ToCellRelation< array2d< localIndex > > elem2dToElems = compute2dElemToElems( faceData, elemToFaces );
```

Comment:
maybe, since in the code they will eventually be called `faceElements` it would be better to refer to them as such even here. `elem2dToElems` -> `faceElementToElems`

Reply:
While I get your point, currently the [`FaceBlockAPI`](https://github.com/GEOSX/GEOSX/blob/develop/src/coreComponents/mesh/generators/FaceBlockABC.hpp) uses the `2dElems` everywhere. So I wanted to be consistent in that regard?
Don't you think it would be better to make all the renaming at once?
Because we don't know when we'll be able to do it... not if it's gonna be the selected wording: if `faceElement` looks nice, what about `edgeFace`? Is it better than `2dFace`?

Reply:
Yes, it would be better to do it at once. I am also not sure what the best names are for the other things like `2dFace`, etc... I was just trying to think of something easy coz I personally find it a bit hard while reviewing this PR. I keep forgetting what a 2dFace actually is. But, TBH, this 2d elements concept is confusing in itself. 

In general, this function `importFracture` (and in general this entire file) seems good to me.